app/src/main/java/com/andrew/tasky/agenda/data/reminder/ReminderRepositoryImpl.kt

```diff
@@ -88,7 +90,7 @@ class ReminderRepositoryImpl @Inject constructor(
                     is Resource.Error -> {
                         Log.e(
                             "uploadCreateAndUpdateModifiedReminders error",
-                            result.message ?: "Unknown Error"
+                            result.message?.asString(appContext) ?: "Unknown Error"
```

Comment:
For error logging, you could also extend `UiText` with a function which returns the value without context if it's a dynamic string (which error messages relevant for logs would usually be). What I mean:
```
// Inside UiText.kt
fun tryGetValue(): String? {
    return if(this is DynamicString) value else null
}
```
Then you don't need to pass the app context around in your repos just for the sake of logging. But this is optional, not really something you have to do